keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -422,28 +413,84 @@ required) or even code snippets. If there's any ambiguity about HOW your
+
 ```
-<<[UNRESOLVED]>>
-Effect of different identifier styles (mainly UUID vs human readable) on DNS
-<<[/UNRESOLVED]>>
+# An example object of `id.k8s.io ClusterClaim`:
+
+apiVersion: multicluster.k8s.io/v1
+kind: ClusterClaim
+metadata:
+  name: id.k8s.io
+spec:
+  value: 721ab723-13bc-11e5-aec2-42010af0021e
 ```
 
 #### `clusterset.k8s.io ClusterClaim`
 
+A cluster in a ClusterSet is expected to be authoritatively associated with that ClusterSet by an external process and storage mechanism with a purview above the cluster local boundary, whether that is a cluster registry or just a human running kubectl. (The details of any specific mechanism is out of scope for the MCS API and this KEP -- see the Non-Goals section.) Mirroring this information in the cluster-local `ClusterClaim` CRD will necessarily need to be managed above the level of the cluster itself, since the properties of `clusterset.k8s.io` extend beyond the boundaries of a single cluster, and will likely be something that has access to the cluster registry. It is expected that the mcs-controller ([as described in the MCS API KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api#proposal)), will act as an admission controller to verify individual objects of this claim.
+
+Because there are obligations of the `id.k8s.io ClusterClaim` that are not meanigfully verifiable until a cluster tries to join a ClusterSet and set its `clusterset.k8s.io ClusterClaim`, the admission controller responsible for setting a `clusterset.k8s.io ClusterClaim` will need the ability to reject such an attempt when it is invalid, and alert `[UNRESOLVED]` or possibly affect changes to that cluster's `id.k8s.io ClusterClaim` to make it valid `[/UNRESOLVED]`. Two symptomatic cases of this would be:
+
+1. When a cluster with a given `id.k8s.io ClusterClaim` ties to join a ClusterSet, but a cluster with that same `id.k8s.io ClusterClaim` appears to already be in the set.
+2. When a cluster that does not have a `id.k8s.io ClusterClaim` tries to join a ClusterSet.
+
+In situations like these, the admission controller will need to fail to add the invalid cluster to the ClusterSet by refusing to set its `clusterset.k8s.io ClusterClaim`, and surface an error that is actionable to make the claim valid.
+
 ```
-<<[UNRESOLVED]>>
-How do we associate a cluster with a Clusterset?
-<<[/UNRESOLVED]>>
+# An example object of `clusterset.k8s.io ClusterClaim`:
+
+apiVersion: multicluster.k8s.io/v1
```

Comment:
we might need to workshop this API group name :)

Reply:
From my notes from the last meeting the serious contenders I've got are:
- about.k8s.io
- properties.k8s.io
- clusterproperties.k8s.io
- id.k8s.io

For now I'm going to plan to be bland and expect to switch it to clusterproperties.k8s.io (to match the name of the only kind we currently expect to be in here, and based on a comment at SIG-MC that having the word 'cluster' in there somewhere was worth it to disambiguate it) but am leaving this comment open for more discussion 👀